compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1603,6 +1612,33 @@ private ParserRuleContext getNextRuleForVarName(int nextLookahead) {
         }
     }
 
+    /**
+     * Check whether the given token kind is a compound binary operator.
+     * 
+     * @param kind STToken kind
+     * @return <code>true</code> if the token kind refers to a binary operator. <code>false</code> otherwise
+     */
+    private boolean isCompoundBinaryOperator(SyntaxKind kind) {
+        switch (kind) {
+            case PLUS_TOKEN:
+            case MINUS_TOKEN:
+            case SLASH_TOKEN:
+            case ASTERISK_TOKEN:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    //return whether we choose a assignment op or a compund binary one
+    private ParserRuleContext compoundOrNot(STToken token) {
+        if (isCompoundBinaryOperator(token.kind)) {
```

Comment:
Once we fix the above comment, this method will be used at only one place.
So we can write this logic then and there